src/lib-ssl-iostream/iostream-openssl.c

```diff
@@ -160,6 +160,17 @@ openssl_iostream_set(struct ssl_iostream *ssl_io,
 			return -1;
 		}
 	}
+#if OPENSSL_VERSION_NUMBER >= 0x10002000L
+	if (set->curve_list != NULL && strlen(set->curve_list) > 0 &&
+		strcmp(ctx_set->curve_list, set->curve_list) != 0) {
```

Comment:
Good point... fixed. At least I assume that ctx_set->curve_list being NULL should behave like if the strings are not equal.